File: frontend/src/types/resident.ts
export interface ResidentDto {
  residentId: number | null;
  caseControlNo: string | null;
  internalCode: string | null;
  safehouseId: number | null;
  caseStatus: string | null;
  sex: string | null;
  dateOfBirth: string | null;
  birthStatus: string | null;
  placeOfBirth: string | null;
  religion: string | null;
  caseCategory: string | null;
  subCatOrphaned: string | null;
  subCatTrafficked: string | null;
  subCatChildLabor: string | null;
  subCatPhysicalAbuse: string | null;
  subCatSexualAbuse: string | null;
  subCatOsaec: string | null;
  subCatCicl: string | null;
  subCatAtRisk: string | null;
  subCatStreetChild: string | null;
  subCatChildWithHiv: string | null;
  isPwd: string | null;
  pwdType: string | null;
  hasSpecialNeeds: string | null;
  specialNeedsDiagnosis: string | null;
  familyIs4ps: string | null;
  familySoloParent: string | null;
  familyIndigenous: string | null;
  familyParentPwd: string | null;
  familyInformalSettler: string | null;
  dateOfAdmission: string | null;
  ageUponAdmission: string | null;
  presentAge: string | null;
  lengthOfStay: string | null;
  referralSource: string | null;
  referringAgencyPerson: string | null;
  dateColbRegistered: string | null;
  dateColbObtained: string | null;
  assignedSocialWorker: string | null;
  initialCaseAssessment: string | null;
  dateCaseStudyPrepared: string | null;
  reintegrationType: string | null;
  reintegrationStatus: string | null;
  initialRiskLevel: string | null;
  currentRiskLevel: string | null;
  dateEnrolled: string | null;
  dateClosed: string | null;
  createdAt: string | null;
  notesRestricted: string | null;
}

export interface ResidentCreate {
  caseControlNo?: string;
  internalCode?: string;
  safehouseId?: number;
  caseStatus?: string;
  sex?: string;
  dateOfBirth?: string;
  birthStatus?: string;
  placeOfBirth?: string;
  religion?: string;
  caseCategory?: string;
  subCatOrphaned?: string;
  subCatTrafficked?: string;
  subCatChildLabor?: string;
  subCatPhysicalAbuse?: string;
  subCatSexualAbuse?: string;
  subCatOsaec?: string;
  subCatCicl?: string;
  subCatAtRisk?: string;
  subCatStreetChild?: string;
  subCatChildWithHiv?: string;
  isPwd?: string;
  pwdType?: string;
  hasSpecialNeeds?: string;
  specialNeedsDiagnosis?: string;
  familyIs4ps?: string;
  familySoloParent?: string;
  familyIndigenous?: string;
  familyParentPwd?: string;
  familyInformalSettler?: string;
  dateOfAdmission?: string;
  ageUponAdmission?: string;
  referralSource?: string;
  referringAgencyPerson?: string;
  assignedSocialWorker?: string;
  initialCaseAssessment?: string;
  reintegrationType?: string;
  reintegrationStatus?: string;
  initialRiskLevel?: string;
  currentRiskLevel?: string;
}

export interface ResidentFilters {
  safehouseId?: number;
  caseStatus?: string;
  riskLevel?: string;
  socialWorker?: string;
}

/** Normalize 'Yes'/'No'/'True'/'False' strings from backend to boolean */
export function isTruthy(v: string | null | undefined): boolean {
  return v === 'Yes' || v === 'True' || v === 'true' || v === '1';
}

File: backend/Intex2.API/Intex2.API/Data/Resident.cs
using System;
using System.Collections.Generic;

namespace Intex2.API.Data;

public partial class Resident
{
    public int? ResidentId { get; set; }

    public string? CaseControlNo { get; set; }

    public string? InternalCode { get; set; }

    public int? SafehouseId { get; set; }

    public string? CaseStatus { get; set; }

    public string? Sex { get; set; }

    public string? DateOfBirth { get; set; }

    public string? BirthStatus { get; set; }

    public string? PlaceOfBirth { get; set; }

    public string? Religion { get; set; }

    public string? CaseCategory { get; set; }

    public string? SubCatOrphaned { get; set; }

    public string? SubCatTrafficked { get; set; }

    public string? SubCatChildLabor { get; set; }

    public string? SubCatPhysicalAbuse { get; set; }

    public string? SubCatSexualAbuse { get; set; }

    public string? SubCatOsaec { get; set; }

    public string? SubCatCicl { get; set; }

    public string? SubCatAtRisk { get; set; }

    public string? SubCatStreetChild { get; set; }

    public string? SubCatChildWithHiv { get; set; }

    public string? IsPwd { get; set; }

    public string? PwdType { get; set; }

    public string? HasSpecialNeeds { get; set; }

    public string? SpecialNeedsDiagnosis { get; set; }

    public string? FamilyIs4ps { get; set; }

    public string? FamilySoloParent { get; set; }

    public string? FamilyIndigenous { get; set; }

    public string? FamilyParentPwd { get; set; }

    public string? FamilyInformalSettler { get; set; }

    public string? DateOfAdmission { get; set; }

    public string? AgeUponAdmission { get; set; }

    public string? PresentAge { get; set; }

    public string? LengthOfStay { get; set; }

    public string? ReferralSource { get; set; }

    public string? ReferringAgencyPerson { get; set; }

    public string? DateColbRegistered { get; set; }

    public string? DateColbObtained { get; set; }

    public string? AssignedSocialWorker { get; set; }

    public string? InitialCaseAssessment { get; set; }

    public string? DateCaseStudyPrepared { get; set; }

    public string? ReintegrationType { get; set; }

    public string? ReintegrationStatus { get; set; }

    public string? InitialRiskLevel { get; set; }

    public string? CurrentRiskLevel { get; set; }

    public string? DateEnrolled { get; set; }

    public string? DateClosed { get; set; }

    public string? CreatedAt { get; set; }

    public string? NotesRestricted { get; set; }
}
